pkg/mdformatter/linktransformer/validator.go

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Bartłomiej Płotka @bwplotka
+// Licensed under the Apache License 2.0.
+
+package linktransformer
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	gitHubAPIURL = "https://api.github.com/repos/%v/%v?sort=created&direction=desc&per_page=1"
+)
+
+type GitHubResponse struct {
+	Number int `json:"number"`
+}
+
+type URLValidator struct {
+	Matched bool
+}
+
+// Match link with any one of provided validators.
+func (v Config) GetValidatorForURL(url string) URLValidator {
+	u := URLValidator{Matched: false}
+	for _, val := range v.Validators {
+		if val._regex.MatchString(url) {
```

Comment:
Let's invert. Prefer keep the complex logic with as small indent as possible. We found invert here and continue if not matched avoiding one indent. Same below.

See: https://thanos.io/tip/contributing/coding-style-guide.md/#control-structure-prefer-early-returns-and-avoid-else

Reply:
Alright! Have made the same mistake in a number of places. Will rectify!